src/parser/mod.rs

```diff
@@ -3095,15 +3201,29 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse an multi-dimension array accessing like `[1:3][1][1]`
+    ///
+    /// Parser is right after the first `[`
```

Comment:
given this is a pub function it would be nice if the api was symmetric so that the function also consumed the leading `[`? I think within the parser itself that would mean calling `prev_token()` or replacing with `peek()` before invoking this function which would be a fair price to pay?

Reply:
Make sense.

Reply:
fixed by [67cd877](https://github.com/apache/datafusion-sqlparser-rs/pull/1551/commits/67cd8778224439aa2f1ddd6605fe15ed0cf12279)